tasks/install.yml

```diff
@@ -52,14 +52,11 @@
   become: true
   when: not vault_configuration.disable_mlock
 
-- name: Generate systemd file
-  template:
-    src: vault.service.j2
-    dest: "/etc/systemd/system/vault.service"
-    owner: "{{ vault_user }}"
-    group: "{{ vault_group }}"
-    mode: 0644
-  become: true
+- include: setup-RedHat-6.yml
+  when: ansible_distribution_major_version == '6'
+
+- include: setup-RedHat-7.yml
+  when: ansible_distribution_major_version == '7'
 
```

Comment:
Ah, this makes more sense with context. Let me add the necessary conditions. Any other ones I should take into account?